### PR TITLE
Add channel strategy

### PR DESCRIPTION
**Pull request by pcarranza**
This allows simplifying the built-in commands by removing the check of the command being called over IM or not.

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -16,19 +16,23 @@ const AdminGroup = "admin"
...
+	Check(meeseeks.Request, CommandAuthorization) error
 }
 
 // ErrUserNotAllowed is the error returned when the auth check fails because the user is not in an allowed group
-var ErrUserNotAllowed = errors.New("User no allowed")
+var ErrUserNotAllowed = errors.New("user no allowed")
 
 // ErrChannelNotAllowed is the error returne when the auth check fails because the command was invoked in a not allowed channel
-var ErrChannelNotAllowed = errors.New("Command not allowed in channel")
+var ErrChannelNotAllowed = errors.New("command not allowed in channel")
```
Comment (omame) [inline auth/auth.go]: Lowercase or uppercase? Decide and stick with it. 😄

File: auth/auth.go [hunk 2]
```diff
@@ -16,19 +16,23 @@ const AdminGroup = "admin"
...
 
 // Authorizer is the interface used to check if a user is allowed to run a command
 type Authorizer interface {
-	Check(string, CommandAuthorization) error
+	Check(meeseeks.Request, CommandAuthorization) error
 }
 
 // ErrUserNotAllowed is the error returned when the auth check fails because the user is not in an allowed group
-var ErrUserNotAllowed = errors.New("User no allowed")
+var ErrUserNotAllowed = errors.New("user no allowed")
```
Comment (omame) [inline auth/auth.go]: Lowercase or uppercase? Decide and stick with it. 😄

File: auth/auth.go [hunk 3]
```diff
@@ -16,19 +16,23 @@ const AdminGroup = "admin"
...
 
 // ErrChannelNotAllowed is the error returne when the auth check fails because the command was invoked in a not allowed channel
-var ErrChannelNotAllowed = errors.New("Command not allowed in channel")
+var ErrChannelNotAllowed = errors.New("command not allowed in channel")
+
+// ErrOnlyIMAllowed is the error returne when the auth check fails because the command was invoked on a public channel
+var ErrOnlyIMAllowed = errors.New("Command only allowed in IM")
```
Comment (omame) [inline auth/auth.go]: Lowercase or uppercase? Decide and stick with it. 😄

Comment (pcarranza) [reply auth/auth.go]: arg... I can use some consistency here.

Will send that to a different PR.

In a nutshell, we should use lowercase for errors so it's easy to concatenate them, I thought I fixed this in some other PR.

**Pull request merged**